### PR TITLE
Ensure that dataframe is not mutated on sort_column_value_order.

### LINKED ISSUE #1008: [BUG] Unexpected mutation of original df by `sort_column_value_order`
# Brief Description
The `sort_column_value_order` function adds a new column `cond_order` to the original dataframe (see [here](https://github.com/pyjanitor-devs/pyjanitor/blob/9d3653f959f11150bcc9aca87476efc72affc60a/janitor/functions/sort_column_value_order.py#L55)) for implementation purposes.

But this is mutating the original dataframe in a non-expected way -- the end user shouldn't be seeing this new column since `cond_order` is only used in the internal implementation of the sorting function.

# Minimally Reproducible Code

```python
df = pd.DataFrame({
    "a1": [4, 2, 9, 12],
    "a2": ["x", "y", "z", "y"],
})
df.sort_column_value_order("a2", {"x": 3, "y": 2, "z": 1});
df

# original df now has an additional column...
   a1 a2  cond_order
0   4  x           3
1   2  y           2
2   9  z           1
3  12  y           2
```

### PR DESCRIPTION
**Pull request by ericmjl**
Closes #1008.

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1009--pyjanitor.netlify.app

File: janitor/functions/sort_column_value_order.py
Comment (thatlittleboy) [inline janitor/functions/sort_column_value_order.py]: Just a minor suggestion to make the description of the output df a little clearer / accurate.
```suggestion
    :return: A sorted pandas DataFrame.
```

File: janitor/functions/sort_column_value_order.py
Comment (thatlittleboy) [inline janitor/functions/sort_column_value_order.py]: A couple of improvements I think we can incorporate here:
- Since we are already doing a copy here, we can avoid the use of `new_df` later on, and just use `df` all the way.
- Actually I think this is one example where guard clauses really shine; -> I suggest we do all the relevant type & value checks (using our `check` and `check_column` util functions) and raise all the errors we can first, *before* doing the `df.copy()` and the rest of the sorting logic in an unnested block. This way, we avoid copying df if the input is invalid.
- And the `del new_df['cond_order']` line can be refactored out of the if-else block, to reduce code duplication.

**Review by thatlittleboy — CHANGES_REQUESTED** [review]
I think this is one example where we can put the recently-mentioned guard clause pattern to good use! :)

File: janitor/functions/sort_column_value_order.py
Comment (ericmjl) [reply janitor/functions/sort_column_value_order.py]: Great point, thanks @thatlittleboy!

**Comment by ericmjl** [comment]
Great catches, @thatlittleboy, thanks for the review. If you think the PR is good, please feel free to do the honors!

File: janitor/functions/sort_column_value_order.py
Comment (thatlittleboy) [inline janitor/functions/sort_column_value_order.py]: The way to implement guard clause for this particular if-else check will be something like this:

```python
check("column_value_order", column_value_order, [dict])  # I would probably just throw this in just in case
if not column_value_order:  # once we ensure this var is a dict, a "not var" is the pythonic way to check it is empty
    raise ValueError("column_value_order dictionary cannot be empty")
# then the remainder of the logic here
```

File: janitor/functions/sort_column_value_order.py
Comment (thatlittleboy) [inline janitor/functions/sort_column_value_order.py]: Good call on using assign here :)

File: janitor/functions/sort_column_value_order.py
Comment (thatlittleboy) [inline janitor/functions/sort_column_value_order.py]: Ah, no, this is probably a bad idea. (see [here](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)).

In general, the safer pattern is to initialize with `columns=None` and `if columns is None: columns = []` in the function body. This ensures that `columns`, if not provided by the user during the function call, is always the empty list every function call.

In this case, I think we can probably leave it as it was originally (separate if-else blocks), or have something like:
```python
columns = [] if columns is None else columns
```
and do what you did. I'll leave it to you to decide which you think is cleaner :)

**Review by thatlittleboy — CHANGES_REQUESTED** [review]
There is just one more common-gotcha with regards to mutable default arguments that we have to fix (I think will cause serious bugs if we don't address this).

Hope you understand!

File: janitor/functions/sort_column_value_order.py
Comment (ericmjl) [reply janitor/functions/sort_column_value_order.py]: Good call! I'm going to use the latter.

Comment (ericmjl) [reply janitor/functions/sort_column_value_order.py]: Great catch! Made the change.

File: janitor/functions/sort_column_value_order.py
Comment (ericmjl) [reply janitor/functions/sort_column_value_order.py]: Great catch, thanks @thatlittleboy!

**Review by thatlittleboy — APPROVED** [review]
LGTM now, thanks for the updates! (There seems to be a codecov warning though, I suppose we should address it first before merging, yeah?)

**Comment by ericmjl** [comment]
Just bumped up the code diff coverage!

**Pull request merged**
